ast/src/analyzed/mod.rs

```diff
@@ -769,6 +808,39 @@ pub enum AlgebraicExpression<T> {
     UnaryOperation(AlgebraicUnaryOperator, Box<AlgebraicExpression<T>>),
 }
 
+impl<T> AlgebraicExpression<T> {
+    /// Returns an iterator over all (top-level) expressions in this expression.
+    /// This specifically does not implement Children so that we can implement
+    /// ExpressionVisitable generically.
```

Comment:
```suggestion
    /// This specifically does not implement Children because otherwise it would
    /// have a wrong implementation of ExpressionVisitable (which is implemented
    /// generically for all types that implement Children<Expr>).
```

I think something like this would have helped me.